pytest 9.0 errors instead of skipping when `unittest.SkipTest` is raised

This code:

```
import unittest

def test_skiptest():
    raise unittest.SkipTest()
```

had this behavior on pytest <9.0:

```
$ rm venv -r
$ python3 -m venv ./venv
$ ./venv/bin/pip install "pytest<9"
[...]
Successfully installed iniconfig-2.3.0 packaging-25.0 pluggy-1.6.0 pygments-2.19.2 pytest-8.4.2
$ ./venv/bin/pytest test.py
=== test session starts ===
platform linux -- Python 3.13.5, pytest-8.4.2, pluggy-1.6.0
rootdir: /home/dev-irc
collected 1 item
test.py s        [100%]

=== 1 skipped in 0.03s ===
```

and this on pytest 9.0.0:

```
$ rm venv -r
$ python3 -m venv ./venv
$ ./venv/bin/pip install "pytest==9.0.0"
[...]
Successfully installed iniconfig-2.3.0 packaging-25.0 pluggy-1.6.0 pygments-2.19.2 pytest-9.0.0
$ ./venv/bin/pytest test.py
=== test session starts ===
platform linux -- Python 3.13.5, pytest-9.0.0, pluggy-1.6.0
rootdir: /home/dev-irc
collected 1 item
test.py F       [100%]

==== FAILURES ===
____ test_skiptest ____

    def test_skiptest():
>       raise unittest.SkipTest()
E       unittest.case.SkipTest

test.py:3: SkipTest
=== short test summary info ===
FAILED test.py::test_skiptest - unittest.case.SkipTest
=== 1 failed in 0.06s ===
```

Python version: 3.13.5

--- a/changelog/13895.bugfix.rst
+++ b/changelog/13895.bugfix.rst
@@ -1,0 +1,1 @@
+Restore support for skipping tests via ``raise unittest.SkipTest``.

--- a/src/_pytest/unittest.py
+++ b/src/_pytest/unittest.py
@@ -476,6 +476,14 @@ def pytest_runtest_makereport(item: Item, call: CallInfo[None]) -> None:
             except AttributeError:
                 pass
 
+    # Convert unittest.SkipTest to pytest.skip.
+    # This covers explicit `raise unittest.SkipTest`.
+    unittest = sys.modules.get("unittest")
+    if unittest and call.excinfo and isinstance(call.excinfo.value, unittest.SkipTest):
+        excinfo = call.excinfo
+        call2 = CallInfo[None].from_call(lambda: skip(str(excinfo.value)), call.when)
+        call.excinfo = call2.excinfo
+
 
 def _is_skipped(obj) -> bool:
     """Return True if the given object has been marked with @unittest.skip."""

--- a/testing/test_unittest.py
+++ b/testing/test_unittest.py
@@ -1094,6 +1094,49 @@ def test_usefixtures_marker_on_unittest(base, pytester: Pytester) -> None:
     result.assert_outcomes(passed=2)
 
 
+def test_skip_setup_class(pytester: Pytester) -> None:
+    """Skipping tests in a class by raising unittest.SkipTest in `setUpClass` (#13985)."""
+    pytester.makepyfile(
+        """
+        import unittest
+
+        class Test(unittest.TestCase):
+
+            @classmethod
+            def setUpClass(cls):
+                raise unittest.SkipTest('Skipping setupclass')
+
+            def test_foo(self):
+                assert False
+
+            def test_bar(self):
+                assert False
+        """
+    )
+    result = pytester.runpytest()
+    result.assert_outcomes(skipped=2)
+
+
+def test_unittest_skip_function(pytester: Pytester) -> None:
+    """
+    Ensure raising an explicit unittest.SkipTest skips standard pytest functions.
+
+    Support for this is debatable -- technically we only support unittest.SkipTest in TestCase subclasses,
+    but stating this support here in this test because users currently expect this to work,
+    so if we ever break it we at least know we are breaking this use case (#13985).
+    """
+    pytester.makepyfile(
+        """
+        import unittest
+
+        def test_foo():
+            raise unittest.SkipTest('Skipping test_foo')
+        """
+    )
+    result = pytester.runpytest()
+    result.assert_outcomes(skipped=1)
+
+
 def test_testcase_handles_init_exceptions(pytester: Pytester) -> None:
     """
     Regression test to make sure exceptions in the __init__ method are bubbled up correctly.

